src/main/java/com/fasterxml/jackson/databind/EnumNamingStrategies.java

```diff
@@ -43,87 +134,338 @@ private EnumNamingStrategies() {}
+            return PropertyNamingStrategies.KebabCaseStrategy.INSTANCE.translate(toBeanName(enumName));
+        }
+    }
+
+    /**
+     * <p>
+     * An implementation of {@link EnumNamingStrategy} that converts enum names in the typical upper
+     * snake case format to lower dot case format.
+     * This implementation first normalizes to lower camel case using (see {@link LowerCamelCaseStrategy} for details)
+     * and then uses {@link PropertyNamingStrategies.LowerDotCaseStrategy} to finish converting the name.
+     * <p>
+     * WARNING: Naming conversion conflicts caused by underscore usage should be handled by client.
+     * e.g. Both <code>PEANUT_BUTTER</code>, <code>PEANUT__BUTTER</code> are converted into "peanut.butter".
+     * And "peanut.butter" will be deserialized into enum with smaller <code>Enum.ordinal()</code> value.
+     *
+     * <p>
+     * This results in the following example conversions:
+     * <ul>
+     * <li>"USER_NAME" is converted into "user.name"</li>
+     * <li>"USER______NAME" is converted into "user.name"</li>
+     * <li>"USERNAME" is converted into "username"</li>
+     * <li>"User__Name" is converted into "user.name"</li>
+     * <li>"_user_name" is converted into "user.name"</li>
+     * <li>"_user_name_s" is converted into "user.name.s"</li>
+     * <li>"__Username" is converted into "username"</li>
+     * <li>"__username" is converted into "username"</li>
+     * <li>"username" is converted into "username"</li>
+     * <li>"Username" is converted into "username"</li>
+     * </ul>
+     *
+     * @since 2.19
+     */
+    public static class LowerDotCaseStrategy implements EnumNamingStrategy {
+
+        /**
+         * An instance of {@link LowerDotCaseStrategy} for reuse.
+         *
+         * @since 2.19
+         */
+        public static final LowerDotCaseStrategy INSTANCE = new LowerDotCaseStrategy();
+
+        @Override
+        public String convertEnumToExternalName(String enumName) {
+            return PropertyNamingStrategies.LowerDotCaseStrategy.INSTANCE.translate(toBeanName(enumName));
+        }
+    }
+
+    /**
+     * Normalizes the enum name to lower camel case in order to be further processed by a PropertyNamingStrategy.
+     *
+     * @param enumName the enum name to be normalized
+     * @return the normalized enum name
+     */
+    private static String toBeanName(String enumName) {
+        if (enumName == null) {
+            return null;
+        }
+
+        final String UNDERSCORE = "_";
+        StringBuilder out = null;
+        int iterationCnt = 0;
+        int lastSeparatorIdx = -1;
+
+        do {
+            lastSeparatorIdx = nextIndexOfUnderscore(enumName, lastSeparatorIdx + 1);
+            if (lastSeparatorIdx != -1) {
+                if (iterationCnt == 0) {
+                    out = new StringBuilder(enumName.length() + 4 * UNDERSCORE.length());
+                    out.append(enumName.substring(iterationCnt, lastSeparatorIdx).toLowerCase());
```

Comment:
Please always pass a `Locale` to `toLowerCase` or there will be locale-specific issues.

Reply:
See `LowerCaseStrategy` and `UpperSnakeCaseStrategy` in `PropertyNamingStrategies`. 
They use toLowerCase() and toUpperCase() without passing in the Locale, which is why I thought I could do the same here.

What Locale should I be passing? Should I update those strategies as well?

Reply:
`Locale.ROOT` is the safe choice. imo the other strategies should use the same

@pjfanning @cowtowncoder WDYT

Reply:
I am not sure; I don't actually like the idea of case change being contextual since that can lead to interoperability problems; similar to using local timezone for date/time values.
But I am not 100% sure how others deal with this.

So how about this: in 2.x, follow existing (not best) Practice of not passing `Locale`.
And then file follow-up issue to change this in 3.0 (`master`) for all casing changes in `jackson-databind`.

Reply:
Actually, reading `Locale` Javadocs, I agree with @yawkat  -- `Locale.ROOT` would make sense.

Any changes to classes outside `EnumNamingStrategy` should go in separate PR tho, fwtw. But assuming safe (`Locale.ROOT` is defaulted to anyway?) could still go in 2.18.

Reply:
I'll change it -- and no, default is "local Locale" otherwise.

But one big challenge if changing to configurable: there's no access to configuration from `EnumNamingStrategy.convertEnumToExternalName()` -- unlike `PropertyNamingStrategy` which takes `MapperConfig`
Should change that for 3.0; will create an issue.